lib/src/screens/home_screen/home_screen.dart

```diff
@@ -1,12 +1,114 @@
+class HomeScreen extends StatefulWidget {
   @override
-  _HomePageState createState() => _HomePageState();
+  _HomeScreenState createState() => _HomeScreenState();
 }
 
-class _HomePageState extends State<HomePage> {
+class _HomeScreenState extends State<HomeScreen> {
   @override
   Widget build(BuildContext context) {
-    return Container();
+    return Scaffold(
+      backgroundColor: ColorsApp.backgroundColor,
+      body: SafeArea(
+        child: Column(
+          children: [
+            Container(
+              height: 200,
+              child: Column(
+                children: [
+                  BlocBuilder<CoinsBloc, CoinsState>(builder: (context, state) {
+                    if (state is CoinsLoadSuccess) {
+                      return SearchBar(state.listCoins!);
+                    }
+                    return SearchBar([]);
+                  }),
+                ],
+              ),
+            ),
+            Expanded(
+              flex: 4,
+              child: Container(
+                width: double.infinity,
+                padding: EdgeInsets.fromLTRB(20, 40, 20, 0),
+                decoration: BoxDecoration(
+                  color: ColorsApp.backgroundBottomColor,
+                  borderRadius: BorderRadius.only(
+                    topRight: Radius.circular(40),
+                    topLeft: Radius.circular(40),
+                  ),
+                ),
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    Padding(
+                      padding: const EdgeInsets.symmetric(vertical: 20),
+                      child: const Text(
+                        StringData.listCoinsTitle,
+                        style: TextStylesApp.listCoinsTitle,
+                      ),
+                    ),
+                    Flexible(
+                      child: BlocBuilder<CoinsBloc, CoinsState>(
+                        builder: (BuildContext context, state) {
+                          if (state is CoinsLoadSuccess) {
+                            return ListView.builder(
+                              itemCount: state.listCoins!.length,
+                              itemBuilder: (BuildContext context, int index) {
+                                return Padding(
+                                  padding:
+                                      const EdgeInsets.symmetric(vertical: 10),
+                                  child: CoinCard(
+                                    image: state.listCoins![index].image!,
+                                    name: state.listCoins![index].name!,
+                                    symbol: state.listCoins![index].symbol!,
+                                    price:
+                                        state.listCoins![index].current_price!,
+                                    price_change: state
+                                        .listCoins![index].price_change_24h!,
```

Comment:
what happens if one of these fields returns a null value?

Reply:
fixed